gdal/ogr/ogrsf_frmts/sqlite/ogrsqliteutility.cpp

```diff
@@ -65,23 +87,13 @@ OGRErr SQLCommand(sqlite3 * poDb, const char * pszSQL)
     return OGRERR_NONE;
 }
 
-OGRErr SQLResultInit(SQLResult * poResult)
-{
-    poResult->papszResult = nullptr;
-    poResult->pszErrMsg = nullptr;
-    poResult->nRowCount = 0;
-    poResult->nColCount = 0;
-    poResult->rc = 0;
-    return OGRERR_NONE;
-}
-
 OGRErr SQLQuery(sqlite3 * poDb, const char * pszSQL, SQLResult * poResult)
 {
     CPLAssert( poDb != nullptr );
     CPLAssert( pszSQL != nullptr );
     CPLAssert( poResult != nullptr );
 
-    SQLResultInit(poResult);
+    poResult->Reset();
```

Comment:
This is the reason to have a Reset function besides constructor/destructor in SQLResult.

I guess we could remove this Reset function if `SQLQuery` was returning a SQLResult instead of taking an existing one. Happy to look into that if you feel that would be better.

Reply:
Great that you tackled this ! Yes, having SQLQuery return a SQLResult would make things more robust

Reply:
Thanks, I'll work on refactoring SQLQuery then !

I have one design question before jumping in: `SQLQuery` can fail and if we can use C++11, I feel the natural way would be to to return a `std::unique_ptr<SQLResult>` that can be nullptr in case of error :

`std::unique_ptr<SQLResult> SQLQuery(sqlite3* poDB, const char* pszSQL)`

This means we loose the error code though (but currently, the only error possibly returned is `OGRERR_FAILURE`, so we aren't loosing much).

If we want to keep the error code, the alternative would be to take a unique_ptr by reference and modify it to store the result if the function returns `OGRERR_NONE`. I'm personally not a big fan of this as it kind of complexifies things I feel because now the caller has to check the returned error code but would also in theory has to check the pointer is actually not null.

`OGRErr SQLQuery(sqlite3* poDB, const char* pszSQL, std::unique_ptr<SQLResult>& poResult)`

Any guidelines about this ? I see the `std::unique_ptr<T>&` pattern used in some places (e.g. `GetCompoundDataType` in netcdfmultidim.cpp), but not sure if its intentional or not.

Reply:
> std::unique_ptr<SQLResult> SQLQuery(sqlite3* poDB, const char* pszSQL)

That signature looks good